### PR TITLE
Refactor UserDeserializer to improve performance

### PR DESCRIPTION
**Pull request by tkrille**
Steps done:
- Make `ObjectMapper` for intermediate result static and reusable
- Remove `ExtensionDeserializer`, moved logic to `UserDeserializer`
- Close `JsonParser` after conversion to `JsonNode`
- User `JsonNode#traverse()` to map intermediate `User`

### CODE REVIEW

**Comment by wallner** [comment]
Please provide more details in the commit message about what this PR does exactly.

File: src/main/java/org/osiam/resources/helper/UserDeserializer.java [hunk 1]
```diff
@@ -77,15 +81,51 @@ public User deserialize(JsonParser jp, DeserializationContext ctxt) throws IOExc
 
             JsonNode extensionNode = rootNode.get(urn);
             if (extensionNode == null) {
-                throw new JsonParseException("Registered extension not present.", JsonLocation.NA);
+                throw new JsonParseException("Registered extension not present: " + urn, JsonLocation.NA);
```
Comment (dacrome) [inline src/main/java/org/osiam/resources/helper/UserDeserializer.java]: String.format?

Comment (wallner) [inline src/main/java/org/osiam/resources/helper/UserDeserializer.java]: Not necessary at this point.

Comment (tkrille) [inline src/main/java/org/osiam/resources/helper/UserDeserializer.java]: String concatenation is sufficient at this point.

**Comment by tkrille** [comment]
Updated.

**Comment by tkrille** [comment]
@wallner Updated again (fixed some typos in commit message).

**Comment by tkrille** [comment]
I have to get used to this fixup thing ... :)

**Comment by wallner** [comment]
Thank you looks good.

**Pull request merged**